### PR TITLE
Event detection

### PR DESCRIPTION
**Pull request by ianlofs**
This does event detection by adding query parameters. An event is an increase in power over some user defined threshold.

### CODE REVIEW

**Comment by ianlofs** [comment]
@iangudger would like your input on this if you have time

File: DB/api/server.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import handler
 
-PORT = 8080
+PORT = 8040
```
Comment (iangudger) [inline DB/api/server.py]: Please revert this change before merging.

Comment (ianlofs) [inline DB/api/server.py]: derp derp derp derp derp

File: DB/api/detect_events.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+				eventList.append(event)
+
+		t5 = t4
+		t4 = t3
+		t3 = t2
+		t2 = t1
+		t1 = float(value)
+		tx = tx + 1
+		center = center + 1
+'''
```
Comment (iangudger) [inline DB/api/detect_events.py]: Please add a newline at the end of this file.

File: DB/api/detect_events.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+			event_list.append(tuple([t4[0], "falling"]))
+
+		t1 = t2
+		t2 = t3
+		t3 = t4
+		t4 = t5
+		t5 = t6
+		t6 = time_series[i]
+		prev_avg = (float(t1[1]) + float(t2[1]) + float(t3[1]) + float(t4[1]) + float(t5[1])) / 5
+		curr_avg = (float(t2[1]) + float(t3[1]) + float(t4[1]) + float(t5[1]) + float(t6[1])) / 5
```
Comment (iangudger) [inline DB/api/detect_events.py]: In Python 3 you don't need to cast to float.

File: DB/api/detect_events.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+	t1 = time_series[1]
+	t2 = time_series[2]
+	t3 = time_series[3]
+	t4 = time_series[4]
+	t5 = time_series[5]
+	t6 = time_series[6]
+
+	prev_avg = (float(t1[1]) + float(t2[1]) + float(t3[1]) + float(t4[1]) + float(t5[1])) / 5
+	curr_avg = (float(t2[1]) + float(t3[1]) + float(t4[1]) + float(t5[1]) + float(t6[1])) / 5
```
Comment (iangudger) [inline DB/api/detect_events.py]: In Python 3 you don't need to cast to float.

Comment (ianlofs) [inline DB/api/detect_events.py]: oh cool didn't know that

File: DB/api/detect_events.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
+from itertools import cycle
+
+
+def detect(time_series, threshold):
+	"""
+	take in a series of time and power readings and return a list of events
```
Comment (iangudger) [inline DB/api/detect_events.py]: The summary typically shouldn't be a restatement of the params and return fields.

File: DB/api/detect_events.py (hunk 4 above)
Comment (iangudger) [inline DB/api/detect_events.py]: In Python 3, the / division operator is floating point division. The // division operator is integer division. This is regardless of the type of the numerator or denominator.

Comment (iangudger) [inline DB/api/detect_events.py]: Even if it was required, you should only do a single cast.

Comment (ianlofs) [inline DB/api/detect_events.py]: so it wort string concat them all together into one big string then divide whatever that is by 5?

Comment (iangudger) [inline DB/api/detect_events.py]: Oh, well it would be better to use a map first so you only have to do it once. In Python 3, maps return a generator.

**Comment by ianlofs** [comment]
@iangudger merging this now, updated to reflected your comments on casting, used map instead.
